tests/unit/parts/plugins/test_matter_sdk_plugin.py

```diff
@@ -0,0 +1,146 @@
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+
+import pytest
+from craft_parts import Part, PartInfo, ProjectInfo
+
+from snapcraft.parts.plugins import MatterSdkPlugin
+
+# The repository where the matter SDK resides.
+MATTER_SDK_REPO = "https://github.com/project-chip/connectedhomeip"
+
+
+@pytest.fixture(autouse=True)
+def part_info(new_dir):
+    yield PartInfo(
+        project_info=ProjectInfo(
+            application_name="test", project_name="test-snap", cache_dir=new_dir
+        ),
+        part=Part("my-part", {}),
+    )
+
+
+def test_get_pull_commands(part_info):
+    properties = MatterSdkPlugin.properties_class.unmarshal(
+        {"matter-sdk-version": "master"}
+    )
+    plugin = MatterSdkPlugin(properties=properties, part_info=part_info)
+
+    sdk_version = properties.matter_sdk_version  # type: ignore
+
+    expected_commands = []
+
+    expected_commands.extend(
+        [
+            "if [ ! -d matter ]; then",
+            "    git init",
+            f"   git remote add origin {MATTER_SDK_REPO}",
+            f"   git fetch --depth 1 origin {sdk_version}",
+            "    git checkout FETCH_HEAD",
+            "fi",
+        ]
+    )
+
+    expected_commands.extend(
+        [
+            "scripts/checkout_submodules.py --shallow --platform linux",
+        ]
+    )
+
+    assert plugin.get_pull_commands() == expected_commands
+
+
+def test_get_build_snaps(part_info):
+    properties = MatterSdkPlugin.properties_class.unmarshal(
+        {"matter-sdk-version": "master"}
+    )
+    plugin = MatterSdkPlugin(properties=properties, part_info=part_info)
+    assert plugin.get_build_snaps() == set()
+
+
+def test_get_build_packages(part_info):
+    properties = MatterSdkPlugin.properties_class.unmarshal(
+        {"matter-sdk-version": "master"}
+    )
+    plugin = MatterSdkPlugin(properties=properties, part_info=part_info)
+    assert plugin.get_build_packages() == {
+        "clang",
+        "cmake",
+        "generate-ninja",
+        "git",
+        "libavahi-client-dev",
+        "libcairo2-dev",
+        "libdbus-1-dev",
+        "libgirepository1.0-dev",
+        "libglib2.0-dev",
+        "libreadline-dev",
+        "libssl-dev",
+        "ninja-build",
+        "pkg-config",
+        "python3-dev",
+        "python3-pip",
+        "python3-venv",
+        "unzip",
+        "wget",
+    }
+
+
+def test_get_build_environment(part_info):
+    properties = MatterSdkPlugin.properties_class.unmarshal(
+        {"matter-sdk-version": "master"}
+    )
+    plugin = MatterSdkPlugin(properties=properties, part_info=part_info)
+
+    assert plugin.get_build_environment() == {}
+
+
+def test_get_build_commands(part_info):
+    properties = MatterSdkPlugin.properties_class.unmarshal(
+        {"matter-sdk-version": "master"}
+    )
+    plugin = MatterSdkPlugin(properties=properties, part_info=part_info)
+
+    expected_commands = []
```

Comment:
Same as above, you'd save ~20 lines of code and gain readability if you create the list all at once.